api/envoy/extensions/filters/http/compressor/v3/compressor.proto

```diff
@@ -42,6 +42,11 @@ message Compressor {
     // "application/xhtml+xml", "image/svg+xml", "text/css", "text/html", "text/plain", "text/xml"
     // and their synonyms.
     repeated string content_type = 3;
+
+    // Runtime flag that controls whether we compress or not body if no content-length and/or transfer-encoding
+    // headers exist. If set to true (default value) we compress body only if either content-length header exists
+    // (and the value is bigger than the minimum one allowed) or transfer-encoding set to chunked.
+    config.core.v3.RuntimeFeatureFlag do_not_compress_if_no_required_headers = 4;
```

Comment:
Does it have to part of the permanent config? I think @alyssawilk suggested moving it to the https://github.com/envoyproxy/envoy/blob/master/source/common/runtime/runtime_features.cc

Reply:
mb i missread her, but my understanding was that it should be protected by RuntimeFlag. as defined here, and by @rojkov comment - it should be disabled by default (so no need to add it into https://github.com/envoyproxy/envoy/blob/master/source/common/runtime/runtime_features.cc#L54). could you folks (@rojkov @alyssawilk) shed a light here please?

Reply:
My understanding is that runtime guards are defined here https://github.com/envoyproxy/envoy/blob/master/source/common/runtime/runtime_features.cc I do not know use cases for defining them in the config proto.